### ISSUE TITLE
Group Blocks on some themes (Dyad 2, Hemingway Re-written, possibly more) have padding issues.

### ISSUE THREAD

**Issue by KirkwallDay** [Automattic/wp-calypso#65739]
### Quick summary

The group block will have padding on some elements preventing content from aligning correctly inside the block. This is easiest to see with a group block and center-aligned column blocks inside of it.

### Steps to reproduce

1. Have a site with the Dyad 2 theme activated.
2. Create a group block with a columns block inside of it. Add some test content.
3. Set the columns to show content center-aligned vertically.
4. In the editor it will look fine.
5. Preview the page in a new tab to see the live site view.

Here is the code of the group block I used:

```
<!-- wp:group {"backgroundColor":"light-gray-blue"} -->
<div class="wp-block-group has-light-gray-blue-background-color has-background"><!-- wp:columns -->
<div class="wp-block-columns"><!-- wp:column {"verticalAlignment":"center"} -->
<div class="wp-block-column is-vertically-aligned-center"><!-- wp:heading -->
<h2>test</h2>
<!-- /wp:heading --></div>
<!-- /wp:column -->

<!-- wp:column {"verticalAlignment":"center"} -->
<div class="wp-block-column is-vertically-aligned-center"><!-- wp:paragraph -->
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eget porta diam. Aenean pharetra, tellus eu fermentum pulvinar, lectus est dignissim quam, vitae tempus nibh ipsum eget ante. Interdum et malesuada fames ac ante ipsum primis in faucibus. Curabitur vitae mollis mauris. In maximus venenatis facilisis. Sed non porta velit, id auctor tellus. Vestibulum id libero maximus, finibus leo ut, facilisis massa. Cras in placerat ante, nec congue dui.</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div>
<!-- /wp:group -->
```

### What you expected to happen

The blocks should look like they do in the editor.

### What actually happened

Instead, there is padding added to the bottom of the block.

Editor view:
<img width="877" alt="Editor View" src="https://user-images.githubusercontent.com/49131029/179781277-adbad4b2-bc07-4ae4-9e54-81fd86194529.png">

Live View:
<img width="856" alt="Dyad 2 Live" src="https://user-images.githubusercontent.com/49131029/179781326-bed53ac8-dc8a-44fd-a8d5-0aa590b4a20e.png">

The space with the red rectangle shows the extra padding added by the theme.

Hemingway Re-written live view:
<img width="905" alt="Hemingway Re-written" src="https://user-images.githubusercontent.com/49131029/179781429-60b5efdf-3930-4993-a42c-535bb0c39209.png">

Less pronounced but still there.

Twenty Twenty-One live view:
<img width="741" alt="Twenty Twenty-two" src="https://user-images.githubusercontent.com/49131029/179781531-31c45932-f45f-4eb5-95d3-5b4dc5b90752.png">

Looks good here.

### Context

- [ ] 5405228-zen

### Platform (Simple, Atomic, or both?)

Simple, Atomic

### Theme-specific issue?

Dyad 2, Hemingway Re-written

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Workaround details

CSS can fix the issue, it seems to vary from theme to theme.

On Dyad 2, set a no-padding custom class to the block and add this code:

```
.no-padding {
padding-top: 1.25em !important;
padding-bottom: 0 !important;
}
```

On Hemingway Re-written it's caused by the paragraph block having added padding. Set the same no-padding class to the group block and then use this code:

```
.no-padding p {
    margin-bottom: 0;
}
```

**Comment by devNigel** [comment]
I tried to reproduce this issue and I noticed that the Group block has a [setting to remove the top and bottom padding](https://d.pr/i/5xpOOM) which worked on Editor view but had no effect on the live page. Was able to confirm with multiple themes like Dyad 2, 2020. 

I think that issue is also related to this.

**Comment by supernovia** [comment]
> The blocks should look like they do in the editor.

@KirkwallDay would you happen to have a handy screenshot of that?

**Comment by renata-franco** [comment]
📌 SCRUBBING
 
- Tested on Simple ✅ 
- Tested on AT ✅ 
- Tested on Self-hosted ✅ 
- Replicable on Core - Yes

------

📌 FINDINGS/SCREENSHOTS/VIDEO

I was able to replicate this issue on Simple and AT sites, as well as self-hosted.

**AT site, Simple site, and Self-Hosted site:**

<table>
<tr>
	<td> Editor Look
	<td> 
<img width="854" alt="imagen" src="https://user-images.githubusercontent.com/81378719/181293246-6093d88f-f3f6-44a0-81ca-4005ac8a93c1.png">

<tr>
	<td> Preview on Desktop
	<td> 
<img width="843" alt="imagen" src="https://user-images.githubusercontent.com/81378719/181293318-5677180d-9125-4e8a-91af-c398326bde4a.png">

</table>



Created my own Group Blocks with two Columns and also three Columns inside. I tested with other blocks outside of the Group Block. 

Checked with the following non-FSE themes and it's not replicable: Rowling, Mayland, Ryu. Could not replicate this using Hemingway Rewritten, which actually looks like this: 

<img width="1139" alt="imagen" src="https://user-images.githubusercontent.com/81378719/181304151-cc9f9974-6995-4560-8e87-d1b09ab35ceb.png">

Regarding [this comment](https://github.com/Automattic/wp-calypso/issues/65739#issuecomment-1189907379) I do not see this option for the Group Block on my end, not sure why. @devNigel Can you let me know if this is available for a particular reason on your end? 

![imagen](https://user-images.githubusercontent.com/81378719/181304419-8f5b7227-81eb-4fa7-ad15-01119714fb0f.png)

Here is what I see: 

<img width="1595" alt="imagen" src="https://user-images.githubusercontent.com/81378719/181305121-4f85c7fe-0ec9-4952-9134-42e9eb48b940.png">



------

📌 ACTIONS

- Marked as Triaged for Quality Squad review

**Comment by devNigel** [comment]
@renata-franco  Do you have CoBlocks active? Based on [this thread](p1658989696450379/1658988071.088579-slack-C03TY6J1A), it looks like it is a feature from CoBlocks.

**Comment by renata-franco** [comment]
> @renata-franco Do you have CoBlocks active? Based on [this thread](https://github.com/Automattic/wp-calypso/issues/p1658989696450379/1658988071.088579-slack-C03TY6J1A), it looks like it is a feature from CoBlocks.

No, I do not have CoBlocks on any of the sites. We stopped having that plugin as default a while ago. Can you try deactivating it on the site where you took the screenshot to see if the option still appears?
